src/processHandler.js

```diff
@@ -0,0 +1,127 @@
+import { auditLogger } from './logger';
+import { sequelize, descriptiveDetails, isConnectionOpen } from './models';
+
+let isShuttingDown = false; // To prevent multiple shutdown attempts
+
+export const resetShutDownFlag = () => {
+  isShuttingDown = false;
+};
+
+export const gracefulShutdown = async (msg) => {
+  if (isShuttingDown) {
+    return;
+  }
+  isShuttingDown = true;
+
+  const details = JSON.stringify(descriptiveDetails());
+  if (isConnectionOpen()) {
+    try {
+      await sequelize.close();
+      auditLogger.info(`Sequelize disconnected through ${msg}: ${details}`);
+    } catch (err) {
+      auditLogger.error(`Error during Sequelize disconnection through ${msg}: ${details}: ${err}`);
+    }
+  } else {
+    auditLogger.info(`Sequelize already disconnected through ${msg}: ${details}`);
+  }
+};
+
+export const formatLogObject = (logObject) => ({
+  message: logObject.message,
+  name: logObject.name,
+  stack: logObject.stack,
+  ...logObject,
+});
+
+export const registerEventListener = () => {
+  // Listen for _fatalException
+  process.on('_fatalException', async (err) => {
+    auditLogger.error('Fatal exception', formatLogObject(err));
+    await gracefulShutdown('fatal exception');
+    process.exit(1);
+  });
+
+  // Listen for uncaught exceptions
+  process.on('uncaughtException', async (err) => {
+    auditLogger.error('Uncaught exception', formatLogObject(err));
+    await gracefulShutdown('uncaught exception');
+    process.exit(1);
+  });
+
+  // Listen for unhandled rejection
+  process.on('unhandledRejection', async (reason, promise) => {
+    auditLogger.error(`Unhandled rejection at: ${promise} reason: ${reason}`);
+
+    if (process.env.CI) {
+      if (reason instanceof Error) {
+        if (reason.message.toLowerCase().includes('maxretriesperrequest')) {
+          return;
+        }
+        auditLogger.error('Uncaught rejection', formatLogObject(reason));
+      }
+    }
+
+    await gracefulShutdown('app termination (unhandledRejection)');
+    process.exit(1);
+  });
+
+  // Listen for SIGINT (Ctrl+C)
+  process.on('SIGINT', async () => {
+    auditLogger.error('Received SIGINT');
+    await gracefulShutdown('app termination (SIGINT)');
+    process.exit(0);
+  });
+
+  // Listen for SIGTERM (e.g., kill command)
+  process.on('SIGTERM', async () => {
+    auditLogger.error('Received SIGTERM');
+    await gracefulShutdown('app termination (SIGTERM)');
+    process.exit(0);
+  });
+
+  // Listen for SIGUSR1
+  process.on('SIGUSR1', async () => {
```

Comment:
Do we have a link to some sort of documentation that goes over what all these various SIG's are?